### DISCUSSION TITLE
Is there a css property sorter?

### DISCUSSION THREAD

**Discussion by otowa1** [biomejs/biome#3167] [Q&A] [answered]
I want a feature that sorts properties like [CSS Declaration Sorter](https://github.com/Siilwyn/css-declaration-sorter/). Is it possible with Biome?

**Comment by ematipico** [comment, marked as answer]
Not yet :)

**Reply by otowa1** [reply]
I'm waiting patiently for it to be added.👌

**Reply by JamyGolden** [reply]
@ematipico is there a stylelint type project on the radar at all? (I’d be interested in contributing)

**Reply by ematipico** [reply]
What do you mean by that? We have 
- https://github.com/biomejs/biome/discussions/2422
- https://github.com/biomejs/biome/issues/2511

Not sure if they answer your question

**Reply by JamyGolden** [reply]
Thanks, yes that answers my question! I had no idea there was work done (let alone so much progress) to add linting/formatting to css :pray: I'll look into contributing

**Reply by vincerubinetti** [reply]
Just want to second the desire for this. I'm surprised more devs don't seem to use that css-declaration-sorting prettier plugin. It really helps me write CSS faster by just writing what comes to mind and not having to think about order. Then the sorting also allows me to easily spot accidental duplicate properties (happens a lot in quickly iterating repos). Also once you get used to it, you know roughly where in the list you need to look for something.

Also, when you get to this, _please_ don't do just alphabetical lol. It makes sense for arbitrarily spelled things like imports, but CSS props need to have some logical grouping. Makes no sense to put `height` far away from `width`, or `bottom` away from `top`. Personally I've gotten to like `smacss` but `concentric` looks good too.

**Reply by ematipico** [reply]
>  I'm surprised more devs don't seem to use that css-declaration-sorting prettier plugin.

I would be surprised of the opposite. Not every brain works the same way, so I find normal that there are people that like formatters and people that don't, and people that like sorting and people that don't. Or people that don't even care.

Which means that it's practically impossible to make everyone happy 😅

**Comment by ianzone** [comment]
I'm using [`stylelint-order`](https://github.com/hudochenkov/stylelint-order) plugin for `stylelint`, and I'd like to replace it with biome.
